Mathlib/Topology/Category/LightProfinite/Basic.lean

```diff
@@ -3,160 +3,492 @@ Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+  map f := f
+
+open scoped Classical in
+instance (S : LightDiagram.{u}) : SecondCountableTopology S.cone.pt := by
+  rw [← TopologicalSpace.Clopens.countable_iff_second_countable]
+  refine @Countable.of_equiv _ _ ?_ (LocallyConstant.equivClopens (X := S.cone.pt))
+  refine @Function.Surjective.countable
+    (Σ (n : ℕ), LocallyConstant ((S.diagram ⋙ FintypeCat.toProfinite).obj ⟨n⟩) (Fin 2)) _ ?_ ?_ ?_
+  · apply @instCountableSigma _ _ _ ?_
+    intro n
+    refine @Finite.to_countable _ ?_
+    refine @Finite.of_injective _ ((S.diagram ⋙ FintypeCat.toProfinite).obj ⟨n⟩ → (Fin 2)) ?_ _
+      LocallyConstant.coe_injective
+    refine @Pi.finite _ _ ?_ _
+    simp only [Functor.comp_obj, toProfinite_obj_toCompHaus_toTop_α]
+    infer_instance
+  · exact fun a ↦ a.snd.comap (S.cone.π.app ⟨a.fst⟩)
+  · intro a
+    obtain ⟨n, g, h⟩ := Profinite.exists_locallyConstant S.cone S.isLimit a
+    exact ⟨⟨unop n, g⟩, h.symm⟩
+
+/-- The inverse part of the equivalence `LightProfinite ≌ LightDiagram` -/
+@[simps]
+def lightDiagramToLightProfinite : LightDiagram.{u} ⥤ LightProfinite.{u} where
+  obj X := LightProfinite.of X.cone.pt
+  map f := f
+
+/-- The equivalence of categories `LightProfinite ≌ LightDiagram` -/
+noncomputable def LightProfinite.equivDiagram : LightProfinite.{u} ≌ LightDiagram.{u} where
+  functor := lightProfiniteToLightDiagram
+  inverse := lightDiagramToLightProfinite
+  unitIso := Iso.refl _
+  counitIso := NatIso.ofComponents
+    (fun X ↦ lightDiagramToProfinite.preimageIso (Iso.refl _)) (by
+      intro _ _ f
+      simp only [Functor.comp_obj, lightDiagramToLightProfinite_obj,
+        lightProfiniteToLightDiagram_obj, Functor.id_obj, Functor.comp_map,
+        lightDiagramToLightProfinite_map, lightProfiniteToLightDiagram_map,
+        lightDiagramToProfinite_obj, Functor.preimageIso_hom, Iso.refl_hom, Functor.id_map]
+      erw [lightDiagramToProfinite.preimage_id, lightDiagramToProfinite.preimage_id,
+        Category.comp_id f])
+  functor_unitIso_comp := by
+    intro
+    simp only [Functor.id_obj, lightProfiniteToLightDiagram_obj, Functor.comp_obj,
+      lightDiagramToLightProfinite_obj, Iso.refl_hom, NatTrans.id_app,
+      lightProfiniteToLightDiagram_map, lightDiagramToProfinite_obj,
+      NatIso.ofComponents_hom_app, Functor.preimageIso_hom]
+    exact lightDiagramToProfinite.preimage_id
+
+instance : lightProfiniteToLightDiagram.IsEquivalence :=
+  show LightProfinite.equivDiagram.functor.IsEquivalence from inferInstance
+
+instance : lightDiagramToLightProfinite.IsEquivalence :=
+  show LightProfinite.equivDiagram.inverse.IsEquivalence from inferInstance
+
 noncomputable section EssentiallySmall
 
-open LightProfinite
+open LightDiagram
 
-/-- This is an auxiliary definition used to show that `LightProfinite` is essentially small. -/
-structure LightProfinite' : Type u where
+/-- This is an auxiliary definition used to show that `LightDiagram` is essentially small. -/
+structure LightDiagram' : Type u where
```

Comment:
Note that `ℕᵒᵖ ⥤ FintypeCat.Skeleton.{u}` already has the structure of a category, but my understanding is that you're giving `LightDiagram'` a different structure (the morphisms are different I think?). Maybe say this in some comment or docstring?